src/osd/osd_types.cc

```diff
@@ -1440,6 +1444,16 @@ void pg_pool_t::decode(bufferlist::iterator& bl)
   } else {
     fast_read = false;
   }
+  if (struct_v >= 23) {
```

Comment:
Since the two flags are added at the same time, maybe only increase the version by one?